src/main/resources/templates/mail/welcome.html

```diff
@@ -1,213 +1,102 @@
+    body{
+      height: 100%;
+      display:grid;
+      justify-content: center;
+      align-items: center;
+      padding: 10px 10px 50px;
+      background: white;
     }
-  }
-
-  body,
-  table,
-  td,
-  a {
-    -ms-text-size-adjust: 100%; /* 1 */
-    -webkit-text-size-adjust: 100%; /* 2 */
-  }
-
-  table,
-  td {
-    mso-table-rspace: 0pt;
-    mso-table-lspace: 0pt;
-  }
-
-  img {
-    -ms-interpolation-mode: bicubic;
-  }
-
-  a[x-apple-data-detectors] {
-    font-family: inherit !important;
-    font-size: inherit !important;
-    font-weight: inherit !important;
-    line-height: inherit !important;
-    color: inherit !important;
-    text-decoration: none !important;
-  }
-
 
-  div[style*="margin: 16px 0;"] {
-    margin: 0 !important;
-  }
-
-  body {
-    width: 100% !important;
-    height: 100% !important;
-    padding: 0 !important;
-    margin: 0 !important;
-  }
+    .logo{
+      display: block;
+      margin-left: auto;
+      margin-right: auto;
+      margin-top: auto;
+    }
 
+    h1 {
+      text-align: center;
+    }
 
-  table {
-    border-collapse: collapse !important;
-  }
+    h2 {
+      text-align: center;
+    }
 
-  a {
-    color: black;
-  }
+    .container{
+      max-width: 800px;
+      width: 100%;
+      background-color: #fff;
+      padding: 25px 30px;
+      border-radius: 10px;
+      box-shadow: 0 0 5px #fcc629;
+    }
 
-  img {
-    height: auto;
-    line-height: 100%;
-    text-decoration: none;
-    border: 0;
-    outline: none;
-  }
+    .buttons:hover{
+      background: #ffc629;
+    }
 
+    .buttons{
+      height: 45px;
+      width: 49%;
+      background-color: #a7a7a7;
+      border-radius: 5px;
+      border: none;
+      margin-left: auto;
+      margin-right: auto;
 
+    }
+    .welcome{
+      max-width: 800px;
+      width: 100%;
+      background-color: #fff;
+      padding: 25px 30px;
+      border-radius: 10px;
+      box-shadow: 0 0 5px #fcc629;
+    }
+   .verify{
+     text-align: center;
+   }
   </style>
-
 </head>
-<body bgcolor="#e9ecef">
 
-  <!-- start preheader -->
-  <div class="preheader" style="display: none; max-width: 0; max-height: 0; overflow: hidden; font-size: 1px; line-height: 1px; color: #fff; opacity: 0;">
-    A preheader is the short summary text that follows the subject line when an email is viewed in the inbox.
+<body>
+<img class="logo" src="/img/logo.png" alt="logo"><br>
+<div class="welcome">
+  <div class="verify">
+    <form action="#"  class="Signature">
+  <p> Hello! <br> Thank you for registering with VirtuDoc! Please click on the button below to verify your
+    email address and get started on connecting you with your ideal physician! </p><br><br>
+      <!---<input class="buttons" type="button" onclick="window.location.href='/HIPPA_consent.html'" value="Verify Your Email"/>-->
+      <a href="/HIPAA_consent.html">Verify</a>

```

Comment:
@aaraujo97 is this the link you mentioned that you were having issues with?

Emails are a little special in terms of what you can and can't do, and also in terms of how links work. Direct links to HTML files don't work inside Thymeleaf templates (well, kind of, but that doesn't apply here). For this to work inside an email client, this needs to be changed to `<a href="https://virtudoc.herokuapp.com/consent">...`. Note that this link will only work once your changes are pushed to production. In the future we can create a URLHelper class that dynamically generates the absolute link based on what environment the application is running in. If you want to test locally, use the URL `https://localhost/consent` or `http://localhost:8080/consent`. As far as why in the `<input...` and `<form...` elements doesn't work, all email clients block both POST requests (typically what HTML `<form...` elements use) and JavaScript (it kind of works in SMTP4dev because it's not a real email client, but it won't work in Outlook or Thunderbird when running on production).

Then to get the consent form to render, you'll need to add a GET route inside the `UserAccountController.java`. Something like this:
```java
@GetMapping("/consent")
public String promptForConsent() {
    return "HIPAA_consent.html";
}
```

As far as how to link the consent request to a user account, there's a couple ways to go about it since you can't use HTML forms. Personally, I would just cheat for now and mark all accounts in the database as consented and activated when the user clicks OK on the consent form. Down the road, we can implement dynamic URL generation in the email and use that, so accounts are individually consented and unlocked. The way that would work is that the email would have a unique link (e.g. `https://virtudoc.herokuapp.com/consent/XXXXXX...`) and the `XXXXX....` part is a VerificationRequest object in the database that links to a UserAccount that gets activated/consented upon visiting the page. That's a complex enough feature that I think we should get this working first, and then iteratively build up to that over the course of several smaller PRs. Up to you guys on how you want to work that in.

Sorry for the wall of text. Feel free to leave a comment if you need clarification on any of the above.

Reply:
Thanks Josh, this is super helpful! Very detailed for sure, Umar and I are gonna be working on getting this through today so will definitely keep you posted on the progress.